### PR TITLE
Fix bug in `check_empty_containers`

### PR DESCRIPTION
**Pull request by ansley**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#63492**

Differential Revision: [D30402749](https://our.internmc.facebook.com/intern/diff/D30402749)

### CODE REVIEW

**Comment by ansley** [comment]
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

**Review by gmagogsfm — COMMENTED** [review]
Good catch. This also reveals a hole in our test cases. Could we add a test case covering this bug to make sure the fix doesn't regress?

**Comment by ansley** [comment]
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

**Comment by gmagogsfm** [comment]
Code looks good. However the test case comment isn't clear for future readers. Also there are some failures in CI, are they real or just flakes?

**Comment by ansley** [comment]
@ansley has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30402749).

File: test/jit/test_isinstance.py [hunk 1]
```diff
@@ -310,3 +310,12 @@ def fn(x: Any):
             x: int = 2
             fn(x)
             self.assertEqual(len(w), 0)
+
+    def test_empty_container_special_cases(self):
+        # Should not throw "Boolean value of Tensor with no values is
+        # ambiguous" error
+        torch._jit_internal.check_empty_containers(torch.Tensor([]))
```
Comment (gmagogsfm) [inline test/jit/test_isinstance.py:317]: Thanks for adding the test case. I am curious how you came across this bug? Maybe we should that use case that leads you to this bug as a test case as well?

Comment (gmagogsfm) [reply test/jit/test_isinstance.py:317]: @ansley ping

If that use case isn't suitable as unit test, I am OK with landing it as is too.

**Review by gmagogsfm — APPROVED** [review]

**Pull request closed without merging**